autorest/azure/token.go

```diff
@@ -168,12 +191,24 @@ func NewServicePrincipalTokenWithSecret(id string, tenantID string, resource str
 		refreshWithin: defaultRefresh,
 		sender:        &http.Client{},
 	}
-	return spt, nil
+	return spt
+}
+
+// NewServicePrincipalTokenFromManualToken creates a ServicePrincipalToken using the supplied token
+func NewServicePrincipalTokenFromManualToken(id string, tenantID string, resource string, token Token) *ServicePrincipalToken {
```

Comment:
That means ~3 extra constructors then. Duplicates of the existing constructors with a suffix that indicates they take a refreshCallback. Or it means breaking the existing constructors.

I agree that the method either needs to be removed or locking added. I'm just not sure which.